### PR TITLE
remove the option to configure the random seed in tests

### PR DESCRIPTION
**Pull request by joelberkeley**

### CODE REVIEW

File: tests/unit/models/test_model_interfaces.py [hunk 1]
```diff
@@ -291,18 +291,20 @@ def test_gpflow_predictor_predict() -> None:
...
+    assert samples.shape == [num_samples, 1, 1]
 
     sample_mean = tf.reduce_mean(samples, axis=0)
     sample_variance = tf.reduce_mean((samples - sample_mean) ** 2)
 
-    npt.assert_allclose(sample_mean, [[6.25]], rtol=0.01)
-    npt.assert_allclose(sample_variance, 1.0, rtol=0.01)
+    linear_error = 1 / tf.sqrt(tf.cast(num_samples, tf.float32))
+    npt.assert_allclose(sample_mean, [[6.25]], rtol=linear_error)
+    npt.assert_allclose(sample_variance, 1.0, rtol=2 * linear_error)
```
Comment (joelberkeley) [inline tests/unit/models/test_model_interfaces.py:307]: variance, being O(y^2), will have error approx twice that of mu iirc

**Review by ChrisMorter — APPROVED** [review]
LGTM
